core/tx_pool.go

```diff
@@ -431,6 +431,7 @@ func (pool *TxPool) loop() {
 				if time.Since(pool.beats[addr]) > pool.config.Lifetime {
 					list := pool.queue[addr].Flatten()
 					for _, tx := range list {
+						log.Info("evict tx for timeout", "tx", tx.Hash().String())
```

Comment:
Why do we want to add these logs (this and the others)? Just for debugging?

Better to add most of these as Debug logs, otherwise there will be too many logs during normal operation.

Reply:
<img width="449" alt="image" src="https://github.com/user-attachments/assets/a4cf1e9f-062c-481d-ab5c-525076d96ac9">

for example, when I investigated the tx life cycle issue, I want know which tx affects this. If we add these logs, we can know what happened with the txs

Reply:
Just want display the tx active state in txpool

Reply:
I also think it's not a good idea to spam `INFO` logs. Then the general logs output when running a normal node is just too overwhelming and one can't make sense out of it anymore. If need be better to move to `DEBUG` and run the nodes that should have create the metrics on debug level

Maybe it's worth thinking about the following (I'm not sure about exact capabilities Prometheus+Grafana): instead of logging this and parsing the logs, maybe we can use Prometheus to store different "events" per tx hash and transform them with some query. This way we can visualize whatever we're interested in and filter the "active tx".

Reply:
`Prometheus+Grafana` doesn't work for it. Metrics just show a trend of an event, they can't show specified info. for example, `txpool_txfifecycle` it just shows tx occupied time at different percentile ratios, but it can't show which tx cost how much, prometheus can't support that. 

If you want to dig some tx why cost so much, I think there are two ways: 1) log, 2) trace. trace need a lot of development. 

> I also think it's not a good idea to spam INFO logs

for this, I think, it's not too bad. if the log really affect the geth's performance, we can downgrade the log level again

Reply:
> the general logs output when running a normal node is just too overwhelming and one can't make sense out of it anymore

This was not addressed in your reply. Our main concern is not performance but comprehension when we're looking at the logs.

Reply:
As experienced, looking at all the logs to see whether the node is normal, is only useful when starting up. After that, we also need to filter the specified log to check.